docs/testing/unit-testing.md

```diff
@@ -107,62 +107,67 @@ describe('when adding a token', () => {
 
 ## Use `it` to specify the desired behavior for the code under test
 
-As each test [should focus on a single aspect of that behavior](#keep-tests-focused), its description should describe that behavior. This description helps to anchor the purpose of the test, understand the intended behavior, and debug differences with the actual behavior that may occur down the road.
+As each test [has to focus on a single aspect of that behavior](#keep-tests-focused), its description must describe that behavior using these guidelines:
 
-Do not repeat the name of the function or method in the name of the test.
-
-Do not use "should" at the beginning of the test name. The official Jest documentation [omits this word from their examples](https://jestjs.io/docs/next/getting-started), and it creates noise when reviewing the list of tests printed after a run.
+1. Start with an active verb in present tense
+2. Focus on what is being tested, not how
+3. Be explicit about the context when needed
+4. Keep names concise but descriptive
+5. Avoid "should", "when", and other unnecessary words
+6. Do not state obvious successful outcomes ("works successfully", "correctly")
+7. Do not list test parameters; describe what they represent instead
 
 ### Examples
 
-🚫
+🚫 Don't
 
 ```typescript
-it('should not stop the block tracker', () => {
+it('should successfully add token when address is valid and decimals are set and symbol exists', () => {
   // ...
 });
-```
-
-✅
 
-```typescript
-it('does not stop the block tracker', () => {
+it('should fail and show error message when invalid address is provided', () => {
   // ...
 });
-```
 
-🚫
+it('works correctly when processing the transaction', () => {
+  // ...
+});
 
-```typescript
-describe('TokensController', () => {
-  it('addToken', () => {
```

Comment:
I wonder if we should still include this example or something like it. This is something I've seen in some really old tests, where people don't bother to write a description for the thing they're testing. Then again, maybe it's obvious, I don't know. Just thought I'd point it out.

Reply:
yes definitely, will work on this